src/contracts/middleware/VoteWeigherBase.sol

```diff
@@ -210,8 +208,12 @@ contract VoteWeigherBase is VoteWeigherBaseStorage {
                     ++j;
                 }
             }
+            require(
+                _newStrategiesConsideredAndMultipliers[i].multiplier > 0,
+                "VoteWeigherBase._addStrategiesConsideredAndMultipliers: cannot add strategy with zero weight"
+            );
```

Comment:
we are requiring the weights be nonzero here but not in `modifyStrategyWeights`
IMO a weight of zero is not invalid, but I feel like we should be consistent about this behavior and clear about what we desire. Probably deserves a dedicated test or two.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/53/commits/c35fa349a17d5bf871aeac38b9df81bb805f9d68 i said a 0 weight is invalid and changed revert messages and added tests. Remove the strategy if you want a zero weight imo.

Reply:
OK, works for me. Great opportunity for a Certora Prover invariant IMO (to verify there's no way to get a 0 multiplier strategy).